include/deal.II/numerics/data_out_resample.h

```diff
@@ -0,0 +1,144 @@
+
+#ifndef dealii_data_out_resample_h
+#define dealii_data_out_resample_h
+
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/data_out_dof_data.h>
+
+#include <memory>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A DataOut class that does not output the result on a numerical solution on
+ * the cells of the original triangulation but interpolates the result onto a
+ * second triangulation (that can be completely unrelated).
+ * By using this class, one can output the result obtained on an unstructured
+ * mesh onto a structured one or one can create a slice in 3D.
+ *
+ * @note While the dimension of the two triangulations might differ, their
+ *   space dimension need to coincide.
+ */
+template <int dim, int patch_dim, int spacedim>
+class DataOutResample
+  : public DataOut_DoFData<dim, patch_dim, spacedim, spacedim>
+{
+public:
+  /**
+   * Constructor taking the triangulation and mapping for which the patches
+   * should be generated.
+   */
+  DataOutResample(const Triangulation<patch_dim, spacedim> &patch_tria,
+                  const Mapping<patch_dim, spacedim> &      patch_mapping);
+
+  /**
+   * Update the @p mapping of original triangulation. One needs to call this
+   * function if the mapping has changed. Just like in the DataOut context,
+   * @p n_subdivisions determines how many "patches" this function will build
+   * out of every cell.
+   *
+   * @note If you use the version of build_patches() that does not take a
+   *   mapping, this function has to be called before its first usage.
+   */
+  void
+  update_mapping(const Mapping<dim, spacedim> &mapping,
+                 const unsigned int            n_subdivisions = 0);
+
+  /**
+   * This is the central function of this class since it builds the list of
+   * patches to be written by the low-level functions of the base class. A
+   * patch is, in essence, some intermediate representation of the data on
+   * each cell of a triangulation and DoFHandler object that can then be used
+   * to write files in some format that is readable by visualization programs.
+   *
+   * Just like in the DataOut context,  @p n_subdivisions determines how many
+   * "patches" this function will build out of every cell.
+   */
+  void
+  build_patches(
+    const Mapping<dim, spacedim> &mapping,
+    const unsigned int            n_subdivisions = 0,
+    const typename DataOut<patch_dim, spacedim>::CurvedCellRegion
+      curved_region =
+        DataOut<patch_dim, spacedim>::CurvedCellRegion::curved_boundary);
+
+  /**
+   * Just like the above function, this function builds a list of
+   * patches to be written by the low-level functions of the base class.
+   * However it skips the update of the mapping and reuses the one registered
+   * via update_mapping(). This allows to skip the expensive setup of the
+   * internal communication routines.
```

Comment:
This would also be useful to state in the documentation of `build_patches()`: That the function re-computes where all of the evaluation nodes are.